gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +207,766 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+
+    :param t: Input MatrixTable or Table.
+    :param info_metrics: List of metrics in info struct of input Table.
+    :param contigs: List of contigs present in input Table.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param delimiter: String to use as the delimiter in XX metrics
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    xx_metrics = [
+        x for x in info_metrics if f"{delimiter}female" in x or f"{delimiter}XX" in x
+    ]
+
+    if "chrY" in contigs:
+        logger.info("Check values of XX metrics for Y variants are NA:")
+        t_y = hl.filter_intervals(t, [hl.parse_locus_interval("chrY")])
+        metrics_values = {}
+        for metric in xx_metrics:
+            metrics_values[metric] = hl.agg.any(hl.is_defined(t_y.info[metric]))
+        output = dict(t_y.aggregate(hl.struct(**metrics_values)))
+        for metric, value in output.items():
+            if value:
+                values_found = t_y.aggregate(
+                    hl.agg.filter(
+                        hl.is_defined(t_y.info[metric]),
+                        hl.agg.take(t_y.info[metric], 1),
+                    )
+                )
+                logger.info(
+                    "FAILED %s = %s check for Y variants. Values found: %s",
+                    metric,
+                    None,
+                    values_found,
+                )
+            else:
+                logger.info("PASSED %s = %s check for Y variants", metric, None)
+
+    t_x = hl.filter_intervals(t, [hl.parse_locus_interval("chrX")])
+    t_xnonpar = t_x.filter(t_x.locus.in_x_nonpar())
+    n = t_xnonpar.count()
+    logger.info("Found %d X nonpar sites", n)
+
+    logger.info("Check (nhomalt == nhomalt_xx) for X nonpar variants:")
+    xx_metrics = [x for x in xx_metrics if "nhomalt" in x]
+
+    field_check_expr = {}
+    field_check_details = {}
+    for metric in xx_metrics:
+        standard_field = metric.replace(f"{delimiter}female", "").replace(
+            f"{delimiter}XX", ""
+        )
+        check_field_left = f"{metric}"
+        check_field_right = f"{standard_field}"
+        field_check_expr, field_check_details = make_field_check_dicts(
+            field_check_expr=field_check_expr,
+            field_check_details=field_check_details,
+            check_description=f"{check_field_left} == {check_field_right}",
+            cond_expr=t_xnonpar.info[check_field_left]
+            != t_xnonpar.info[check_field_right],
+            display_fields=hl.struct(
+                **{
+                    check_field_left: t_xnonpar.info[check_field_left],
+                    check_field_right: t_xnonpar.info[check_field_right],
+                }
+            ),
+        )
+
+    generic_field_check_loop(t_xnonpar, field_check_expr, field_check_details, verbose)
```

Comment:
PART OF LARGE SUGGESTION
```suggestion
        field_check_expr[f"{check_field_left} == {check_field_right}"] = {
            "expr": hl.agg.count_where(
                t_xnonpar.info[check_field_left] != t_xnonpar.info[check_field_right]
            ),
            "display_fields": hl.struct(
                **{
                    check_field_left: t_xnonpar.info[check_field_left],
                    check_field_right: t_xnonpar.info[check_field_right],
                }
            ),
        }

    generic_field_check_loop(t_xnonpar, field_check_expr, verbose)
```